### PR TITLE
Install systemd unit file

### PR DESCRIPTION
**Pull request by ueno**
Also add a Makefile that wraps cargo and install commands.

Fixes: #271
Signed-off-by: Daiki Ueno <dueno@redhat.com>

### CODE REVIEW

File: dist/systemd/system/keylime_agent.service
Comment (sergio-correia) [inline dist/systemd/system/keylime_agent.service:5]: Do we want to specify a (non-root) user here?

Comment (THS-on) [reply dist/systemd/system/keylime_agent.service:5]: Yes it is a good idea to run keylime not as root by default. Example on what we do for the Debian package:
https://github.com/utkarsh2102/python-keylime/blob/master/debian/python3-keylime-agent.keylime_agent.service

Also the mount for `/var/lib/keylime/secure` can then created by systemd:
https://github.com/utkarsh2102/python-keylime/blob/master/debian/python3-keylime-agent.var-lib-keylime-secure.mount

Comment (sergio-correia) [reply dist/systemd/system/keylime_agent.service:5]: @THS-on: nice! Can't (some of) these changes be upstreamed?

Comment (THS-on) [reply dist/systemd/system/keylime_agent.service:5]: @sergio-correia the current installer.sh script just assumes that you run everything as root which is fine because it is mostly used for testing and CI. It is up to the distributions on how to package keylime and if for example they want to create a new user etc.

If there is interest I see no reason why not to include those as an example on how to run Keylime without root.

**Review by lkatalin — COMMENTED** [review]
I'll defer to others on the non-root user issue. I also think it may be good to have a few lines of documentation about these added files and how to invoke the makefile to do different things.

**Review by lkatalin — CHANGES_REQUESTED** [review]
See above

**Comment by lkatalin** [comment]
Related: https://github.com/keylime/keylime/issues/864

File: dist/systemd/system/keylime_agent.service
Comment (lkatalin) [reply dist/systemd/system/keylime_agent.service:5]: @THS-on @sergio-correia @ueno Was I reading this correctly that we are still waiting for a change on this PR re: a non-root user?

Comment (THS-on) [reply dist/systemd/system/keylime_agent.service:5]: No, what the Debian package currently does is not ideal and will not work for packages  actually shipped in distributions. So we can merge it as currently is.

See discussion here: https://github.com/keylime/keylime/issues/864

**Comment by lkatalin** [comment]
> I'll defer to others on the non-root user issue. I also think it may be good to have a few lines of documentation about these added files and how to invoke the makefile to do different things.

@ueno Is it possible to add a few lines of documentation in the `README.md` for these changes?

**Comment by ueno** [comment]
@lkatalin sorry for the delay; added some comments in README.md and keylime_agent.service.

File: dist/systemd/system/keylime_agent.service
Comment (lkatalin) [inline dist/systemd/system/keylime_agent.service:8]:
```suggestion
# and uncomment the following lines:
```

**Comment by lkatalin** [comment]
> @lkatalin sorry for the delay; added some comments in README.md and keylime_agent.service.

@ueno No problem, thank you! Looks great.

**Comment by lkatalin** [comment]
@ueno Sorry I've tried to make one small change in wording without asking you to re-push, but it seems I've mucked it up and can't revert the commit as I don't have force-push access to your branch. Do you think you could re-push it when you get a chance? Then I will be able to merge it.

**Comment by ueno** [comment]
@lkatalin thanks, merged the fix and force-pushed

**Review by lkatalin — APPROVED** [review]
Thank you, @ueno !

**Pull request merged**
